src/utils/testing/mocks/mockUpdate.ts

```diff
@@ -1,19 +1,45 @@
 import dayjs from 'dayjs';
+import { pick } from 'lodash';
 
+import mockJourneyInstance from './mockJourneyInstance';
 import { mockObject } from 'utils/testing/mocks';
 import mockPerson from './mockPerson';
-import { ZetkinUpdate } from 'types/zetkin';
+import { UPDATE_TYPES, ZetkinUpdate } from 'types/updates';
 
-const update: ZetkinUpdate = {
+const update: Partial<ZetkinUpdate> = {
   actor: mockPerson(),
-  created_at: dayjs().subtract(5, 'hours').format(),
-  details: { assignee: mockPerson() },
-  target: { id: 1, title: 'Journey instance' },
-  type: 'journeyInstance.assignee.add',
+  timestamp: dayjs().subtract(5, 'hours').format(),
 };
 
-const mockUpdate = (overrides?: Partial<ZetkinUpdate>): ZetkinUpdate => {
-  return mockObject(update, overrides);
+const mockUpdate = (
+  type: `${UPDATE_TYPES}`,
+  overrides?: Partial<ZetkinUpdate>
+): ZetkinUpdate => {
+  const updateData = {
+    [UPDATE_TYPES.JOURNEYINSTANCE_ADDASSIGNEE]: {
+      details: { assignee: mockPerson() },
+      target: pick(mockJourneyInstance(), ['id', 'title']),
+      type: 'journeyinstance.addassignee',
+    },
+    [UPDATE_TYPES.JOURNEYINSTANCE_UPDATEMILESTONE]: {
+      details: {
+        changes: {
+          completed: { from: null, to: update.timestamp },
+        },
+        milestone: {
+          id: 1,
+          title: 'title of milestone',
+        },
+      },
+      target: pick(mockJourneyInstance(), ['id', 'title']),
+      type: 'journeyinstance.updatemilestone',
+    },
+  };
+
+  return {
+    ...mockObject(update, overrides),
+    ...updateData[type],
```

Comment:
This will alsways override all `overrides` passed in with the update data, so you can't override the data that's used in `updateData`. Maybe you want to instead merge overrides with the `updateData[type]`

```js
return mockObject(update, {...updateData[type, ...overrides})
```

I'm not exactly sure that will work, but then you can override the update info, I think

Reply:
I don't see how that change would make anything "better"

Reply:
I think what @djbusstop is referring to, is that with the current solution any `overrides` provided by the developer will be overridden by whatever is in `updateData[type]`, so that if the developer wants to override the `target` or `details` that's not possible.

If it's not a problem right now it's not a problem, so I don't think it has to change at this time. But I expect it can cause some confusion in the future when someone tries to use the overrides like that and it doesn't work.